plugin/server_audit/server_audit.cc

```diff
@@ -1050,6 +1085,130 @@ static int get_user_host(const char *uh_line, unsigned int uh_len,
 #define S_ISDIR(x) ((x) & _S_IFDIR)
 #endif /*_WIN32 && !S_ISDIR*/
 
+/*
+  Flushes the buffer of log strings to the file.
+
+  @param all_messages Contains the log strings
+*/
+void flush_buffer(const std::vector<std::string>& all_messages) 
```

Comment:
I would push this filename and buffer handling to file_logger.c so that other plugins and/or server code could use buffered logging.